src/streamlink/plugins/hds.py

```diff
@@ -1,22 +1,44 @@
 import re
 
 from streamlink.plugin import Plugin
-from streamlink.plugin.plugin import parse_url_params
+from streamlink.plugin.plugin import parse_url_params, LOW_PRIORITY, NORMAL_PRIORITY, NO_PRIORITY
 from streamlink.stream import HDSStream
 from streamlink.utils import update_scheme
+from streamlink.compat import urlparse
 
 
 class HDSPlugin(Plugin):
-    _url_re = re.compile(r"hds://(.+)")
+    _url_re = re.compile(r"(hds://)?(.+(?:\.f4m)?.*)")
+
+    @classmethod
+    def priority(cls, url):
+        """
+        Returns LOW priority if the URL is not prefixed with hls:// but ends with
+        .m3u8 and return NORMAL priority if the URL is prefixed.
```

Comment:
True that. Took me a while to notice what you were referring to, a more verbose comment would be more helpful ;-)